SCORER.md

```diff
@@ -21,11 +28,278 @@ You may need to build your own scorer - your own _language model_ if:
+     --checkpoint_dir deepspeech-data/checkpoints \
+     --n_hidden 4 \
+     --n_trials 3 \
+     --lm_alpha_max 0.92 \
+     --lm_beta_max 1.05
+```
+
+#### Using `generate_lm.py` to create `lm.binary` and `vocab-500000.txt` files
+
+We then use `generate_lm.py` script that comes with DeepSpeech to create a [_trie file_](https://en.wikipedia.org/wiki/Trie). The _trie file_ represents associations between words, so that during training, words that are more closely associated together are more likely to be transcribed by DeepSpeech.
+
+The _trie file_ is produced using a software package called [KenLM](https://kheafield.com/code/kenlm/). KenLM is designed to create large language models that are able to be filtered and queried easily.
+
+First, create a directory in `deepspeech-data` directory to store your `lm.binary` and `vocab-500000.txt` files:
+
+```
+deepspeech-data$ mkdir indonesian-scorer
+```
+
+Then, use the `generate_lm.py` script as follows:
+
+```
+cd data/lm
+python3 generate_lm.py \
+  --input_txt /DeepSpeech/deepspeech-data/indonesian-sentences.txt \
+  --output_dir /DeepSpeech/deepspeech-data/indonesian-scorer \
+  --top_k 500000 --kenlm_bins /DeepSpeech/native_client/kenlm/build/bin/ \
+  --arpa_order 5 --max_arpa_memory "85%" --arpa_prune "0|0|1" \
+  --binary_a_bits 255 --binary_q_bits 8 --binary_type trie
+```
+
+_Note: the `/DeepSpeech/native_client/kenlm/build/bin/` is the path to the binary files for `kenlm`. If you are using the Docker image and container (explained on the [environment page of the PlayBook](ENVIRONMENT.md)), then `/DeepSpeech/native_client/kenlm/build/bin/` is the correct path to use. If you are not using the Docker environment, your path may vary._
+
+You should now have a `lm.binary` and `vocab-500000.txt` file in your `indonesian-scorer` directory:
+
+```
+deepspeech-data$ ls indonesian-scorer/
+total 1184
+  4 drwxrwxr-x 2 root   root   4096 Feb 25 23:13 ./
+  4 drwxrwxr-x 5 root   root   4096 Feb 26 09:24 ../
+488 -rw-r--r-- 1 root   root      499594 Feb 24 19:05 lm.binary
+ 52 -rw-r--r-- 1 root   root       51178 Feb 24 19:05 vocab-500000.txt
+```
+
+#### Generating a `kenlm.scorer` file from `generate_scorer_package`
+
+Next, we need to install the `native_client` package, which contains the `generate_scorer_package`. This is _not_ pre-built into the DeepSpeech Docker image.
+
+The `generate_scorer_package`, once installed via the `native client` package, is usable on _all platforms_ supported by DeepSpeech. This is so that developers can generate scorers _on-device_, such as on an Android device, or Raspberry Pi 3.
+
+To install `generate_scorer_package`, first download the relevant `native client` package from the [DeepSpeech GitHub releases page](https://github.com/mozilla/DeepSpeech/releases/tag/v0.9.3) into the `data/lm` directory. The Docker image uses Ubuntu Linux, so you should use the `native_client.amd64.cuda.linux.tar.xz` package.
```

Comment:
I'm unsure about cuda there, it's not really related and it might confuse people, pure cpu would work exactly the same

Reply:
Agreed. What I'm trying to do here is provide step by step instructions for the `native client` because it seems to be a frequent stumbling block for people. What I've done is provide instructions for _both_ `cuda` and `cpu`.

Reply:
Just mentionning it since in the past we have had some (on purpose) discrepency on things like that, where we would package something on cpu and not on cuda builds